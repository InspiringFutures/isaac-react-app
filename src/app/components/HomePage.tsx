import React from "react";
import {connect} from "react-redux";

const stateToProps = (state: any) => ({user: state.user});
const dispatchToProps = null;
const HomePageComponent = ({user}: any) => (
    <div>
        <h2>Home</h2>
        <p>Hello {user.givenName}!</p>
    </div>
);
export const HomePage = connect(stateToProps, dispatchToProps)(HomePageComponent);
