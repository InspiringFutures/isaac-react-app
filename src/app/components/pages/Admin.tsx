import React, {useEffect, useState} from 'react';
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import * as RS from "reactstrap";
import {AppState, ContentVersionState} from "../../state/reducers";
import {RegisteredUserDTO} from "../../../IsaacApiTypes";
import {getContentVersion, requestConstantsSegueVersion, setContentVersion} from "../../state/actions";
import {ShowLoading} from "../handlers/ShowLoading";
import {ContentVersionUpdatingStatus} from "../../services/constants";
import {TitleAndBreadcrumb} from "../elements/TitleAndBreadcrumb";

const stateToProps = (state: AppState) => ({
    segueVersion: state && state.constants && state.constants.segueVersion || "unknown",
    contentVersion: state && state.contentVersion || null,
});

const dispatchToProps = {getContentVersion, setContentVersion, requestConstantsSegueVersion};

interface AdminPageProps {
    user: RegisteredUserDTO;
    segueVersion: string;
    contentVersion: ContentVersionState;
    getContentVersion: () => void;
    setContentVersion: (version: string) => void;
    requestConstantsSegueVersion: () => void;
}

const AdminPageComponent = ({user, getContentVersion, setContentVersion, contentVersion, segueVersion, requestConstantsSegueVersion}: AdminPageProps) => {
    useEffect(() => {
        getContentVersion();
    }, []);

    useEffect(() => {
        requestConstantsSegueVersion();
    }, []);

    const [newVersion, setNewVersion] = useState<string | null>(null);

    const displayVersion = newVersion || (contentVersion && contentVersion.liveVersion) || null;

    const startVersionUpdate = function(event?: React.FormEvent) {
        if (event) {
            event.preventDefault();
        }
        if (contentVersion && displayVersion !== contentVersion.liveVersion && newVersion != null) {
            setContentVersion(newVersion);
        }
    };

    const updateState = contentVersion && contentVersion.updateState || null;

    return <RS.Container id="admin-page">
        <TitleAndBreadcrumb currentPageTitle="Isaac administration" breadcrumbTitleOverride="Admin tools" />

        <div className="py-4">

            Hi, {user.givenName}!

            <RS.Card className="p-3 my-3">
                <RS.CardTitle tag="h2">Useful links</RS.CardTitle>
                <RS.CardBody>
                    <Link to="/admin/usermanager">User Manager</Link>
                </RS.CardBody>
            </RS.Card>

            <RS.Card className="p-3 mb-3">
                <RS.CardTitle tag="h2">Admin Console</RS.CardTitle>
                <RS.CardBody>
                    <ul>
                        <li><strong>API Version:</strong> {segueVersion}</li>
                    </ul>
                </RS.CardBody>
            </RS.Card>

            <RS.Card className="p-3 mb-3">
                <RS.CardTitle tag="h2">Administrative tools</RS.CardTitle>
                <RS.CardBody>
                    <h3>Manage site content</h3>
                    {contentVersion && <React.Fragment>
                        <div>
                            <label htmlFor="content-version">
                                <strong>Live Content Version</strong>
                            </label>
                        </div>
                        <ShowLoading until={displayVersion !== null} render={() => {
                            return displayVersion !== null && updateState != ContentVersionUpdatingStatus.UPDATING &&
                                <RS.Form onSubmit={startVersionUpdate}>
                                    <RS.InputGroup>
                                        <RS.Input
                                            type="text" value={displayVersion}
                                            onChange={e => setNewVersion(e.target.value)}
                                            placeholder="Enter commit SHA"
                                        />
                                        <RS.InputGroupAddon addonType="append">
                                            <RS.Button
                                                type="button" className="p-0 border-dark"
                                                onClick={startVersionUpdate}
                                                disabled={displayVersion === contentVersion.liveVersion}
                                            >
                                                Set Version
                                            </RS.Button>
                                        </RS.InputGroupAddon>
                                    </RS.InputGroup>
                                </RS.Form>
                        }} />
                        {updateState == ContentVersionUpdatingStatus.UPDATING &&
                            <RS.Alert color="info">
                                <h4>Updating...</h4>
                                <p>Replacing version {contentVersion.liveVersion} with {contentVersion.updatingVersion}</p>
                                <RS.Spinner />
                            </RS.Alert>
                        }
                        {updateState == ContentVersionUpdatingStatus.SUCCESS &&
                            <RS.Alert color="success">
                                <h4>Content version changed successfully.</h4>
                            </RS.Alert>
                        }
                        {updateState == ContentVersionUpdatingStatus.FAILURE &&
                            <RS.Alert color="danger">
                                <h4>Error: Content version could not be changed.</h4>
                            </RS.Alert>
                        }
                    </React.Fragment>}
                </RS.CardBody>
            </RS.Card>

        </div>
    </RS.Container>;
};

export const Admin = connect(stateToProps, dispatchToProps)(AdminPageComponent);
