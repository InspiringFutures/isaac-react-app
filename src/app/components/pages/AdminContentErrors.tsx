import React, {useEffect} from "react";
import {AdminContentErrorsState, AppState} from "../../state/reducers";
import {getAdminContentErrors} from "../../state/actions";
import {ShowLoading} from "../handlers/ShowLoading";
import {connect} from "react-redux";
import {withRouter} from "react-router-dom";
import {Col, Container, Row, Table} from "reactstrap";
import {EDITOR_URL} from "../../services/constants";
import {ContentErrorItem} from "../../../IsaacAppTypes";
import {TitleAndBreadcrumb} from "../elements/TitleAndBreadcrumb";

const stateToProps = (state: AppState) => {
    return {
        errors: state ? state.adminContentErrors : null
    };
};
const dispatchToProps = {getAdminContentErrors};

interface AdminContentErrorsPageComponentProps {
    errors: AdminContentErrorsState;
    getAdminContentErrors: () => void;
}

const contentErrorDetailsListItem = (errorDetailsListItem: string, index: number) => {
    return <li key={index}>{errorDetailsListItem}</li>
};

const contentErrorRow = (errorRecord: ContentErrorItem, index: number) => {
    return <tr key={index}>
        <td>
            {/* eslint-disable-next-line react/jsx-no-target-blank */}
            <a href={EDITOR_URL + errorRecord.partialContent.canonicalSourceFile} target="_blank" rel="noopener">
                {errorRecord.partialContent.title || errorRecord.partialContent.id}
            </a>
        </td>
        <td>{(!!errorRecord.partialContent.published).toString()}</td>{/*The published key may be missing, also meaning false*/}
        <td>{(!errorRecord.successfulIngest).toString()}</td>
        <td>
            <ul>
                {errorRecord.listOfErrors.map(contentErrorDetailsListItem)}
            </ul>
        </td>
    </tr>
};

export const AdminContentErrorsPageComponent = ({errors, getAdminContentErrors}: AdminContentErrorsPageComponentProps) => {
    useEffect(
        () => {getAdminContentErrors();}, []
    );

    return <Container>
        <Row>
            <Col>
                <TitleAndBreadcrumb currentPageTitle="Content errors" />
            </Col>
        </Row>
        <ShowLoading until={errors}>
            {errors && <div>
                <Row>
                    <Col>
                        <p>
                            <strong>Critical errors:</strong> {errors.failedFiles},&nbsp;
                            <strong>Files with errors:</strong> {errors.brokenFiles},&nbsp;
                            <strong>Total errors:</strong> {errors.totalErrors}
                        </p>
                        <p>
                            <strong>Content Version:</strong> {errors.currentLiveVersion}
                        </p>
                    </Col>
                </Row>
                <Row>
                    <Col>
                        <Table>
                            <tbody>
                                <tr>
                                    <th>Title / Filename</th>
                                    <th title="Is this file published?">Published</th>
                                    <th title="Files with critical errors will not be available on Isaac!">Critical Error</th>
                                    <th>List of Error Messages</th>
                                </tr>
                                {errors.errorsList.map(contentErrorRow)}
                            </tbody>
                        </Table>
                    </Col>
                </Row>
            </div>}
        </ShowLoading>
    </Container>;
};

export const AdminContentErrors = withRouter(connect(stateToProps, dispatchToProps)(AdminContentErrorsPageComponent));
