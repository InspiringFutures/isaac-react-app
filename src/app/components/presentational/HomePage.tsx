import React from "react";

const HomePage = () =>
    <div>
        <h2>Home</h2>
        <p>Hello!</p>
    </div>;

export default HomePage;
