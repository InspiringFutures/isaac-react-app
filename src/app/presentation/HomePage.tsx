import React from "react";

const Home = ({name}: {name: string}) =>
    <div>
        <h2>Home</h2>
        <p>Hello {name}</p>
    </div>;

export default Home;
