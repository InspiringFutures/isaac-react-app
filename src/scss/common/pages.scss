// ISAAC
main {
  min-height: 450px;
}

// NOMENSA pages.scss
// COMMENT OUT THE BACKGROUND GRADIENTS
.question-difficulty {
  background-color: white;
  box-shadow: 0 2px 30px 0 rgba(0, 0, 0, 0.08);
  width: 223px;
  h3 {
    font-size: 1.25rem;
  }
}

.homepageHero {
  //background: rgb(246, 246, 246); /* Old browsers */
  //background: -moz-linear-gradient(
  //                top,
  //                rgba(246, 246, 246, 1) 0%,
  //                rgba(255, 255, 255, 1) 6%
  //); /* FF3.6-15 */
  //background: -webkit-linear-gradient(
  //                top,
  //                rgba(246, 246, 246, 1) 0%,
  //                rgba(255, 255, 255, 1) 6%
  //); /* Chrome10-25,Safari5.1-6 */
  //background: linear-gradient(
  //                to bottom,
  //                rgba(246, 246, 246, 1) 0%,
  //                rgba(255, 255, 255, 1) 6%
  //); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  //filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#f6f6f6', endColorstr='#ffffff',GradientType=0 ); /* IE6-9 */
  //
  button {
    margin-top: 1rem;
  }
}

.homepagePhyTable {
  width: 100%;
  background: $white;
  th {
    width: 40px;
    text-align: center;
    color: white;
    text-transform: uppercase;
    font-size: 19px;
    font-family: $secondary-font-bold;
    font-weight: 700;
    span {
      position: absolute;
      transform: rotate(-90deg);
      transform-origin: 13% 75%;
    }
  }
  td {
    height: 40px;
    position: relative;
    line-height: 40px;
    padding: 0;
  }
  td:hover {
    background: $gray-107;
  }
  a {
    display: block;
    padding-left: 18px;
    text-decoration: none;
  }
  td:hover:before {
    content: '';
    border-top: 20px solid #ececec;
    border-left: 15px solid #ececec;
    border-bottom: 20px solid #ececec;
    display: inline-block;
    position: absolute;
    top: 0;
    left: 0;
    pointer-events: none;
  }
}

.homepageLearn {
  th {
    background: theme-color("primary");
  }
  td:hover:before {
    border-left-color: theme-color("primary");
  }
}

.homepageTeach {
  th {
    background: $black;
  }
  td:hover:before {
    border-left-color: $black;
  }
}