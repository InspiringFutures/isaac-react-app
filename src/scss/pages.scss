.question-difficulty {
  background-color: white;
  box-shadow: 0 2px 30px 0 rgba(0, 0, 0, 0.08);
  width: 223px;
  h3 {
    font-size: 1.25rem;
  }
}
.homepageHero {
  background: rgb(246, 246, 246); /* Old browsers */
  background: -moz-linear-gradient(
                  top,
                  rgba(246, 246, 246, 1) 0%,
                  rgba(255, 255, 255, 1) 6%
  ); /* FF3.6-15 */
  background: -webkit-linear-gradient(
                  top,
                  rgba(246, 246, 246, 1) 0%,
                  rgba(255, 255, 255, 1) 6%
  ); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(
                  to bottom,
                  rgba(246, 246, 246, 1) 0%,
                  rgba(255, 255, 255, 1) 6%
  ); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#f6f6f6', endColorstr='#ffffff',GradientType=0 ); /* IE6-9 */

  button {
    margin-top: 1rem;
  }
}
