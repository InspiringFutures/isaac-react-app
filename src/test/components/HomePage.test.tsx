import React from 'react';
import {shallow} from 'enzyme';
import {HomePageComponent} from "../../app/components/pages/HomePage";
import {RegisteredUserDTO} from "../../IsaacApiTypes";

describe('HomePageComponent', () => {

    it('renders correctly with a null valued user', () => {
        const component = shallow(<HomePageComponent user={null} />);
        expect(component).toMatchSnapshot();
    });

    it("prints out the user's name when it is provided", () => {
        const user: RegisteredUserDTO = {givenName: "Alan"};
        const component = shallow(<HomePageComponent user={user} />);
        expect(component).toMatchSnapshot();
    })

});
