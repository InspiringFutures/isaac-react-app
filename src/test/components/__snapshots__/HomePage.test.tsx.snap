// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageComponent prints out the user's name when it is provided 1`] = `
<div
  id="homepage"
>
  <section
    id="call-to-action"
  >
    <Row
      tag="div"
    >
      <Col
        md={6}
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Row
          tag="div"
        >
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <h1>
              Welcome Alan!
            </h1>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla feugiat lorem nisl, sed convallis dui lobortis volutpat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla feugiat lorem nisl, sed convallis dui lobortis volutpat.
            </p>
          </Col>
        </Row>
        <Row
          tag="div"
        >
          <Col
            size={6}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              block={true}
              color="primary"
              tag={[Function]}
              to="/topics"
            >
              Find a topic
            </Button>
          </Col>
          <Col
            size={6}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              block={true}
              color="dark"
              outline={true}
              tag={[Function]}
              to="/events"
            >
              Find an event
            </Button>
          </Col>
        </Row>
      </Col>
      <Col
        md={6}
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <img
          alt="Students illustration"
          className="img-fluid"
          src="/assets/ics_hero.svg"
        />
      </Col>
    </Row>
  </section>
  <section
    className="px-5 py-4"
    id="headline-content"
  >
    <h3>
      Featured-question
    </h3>
    <Row
      className="p-1"
      tag="div"
    >
      <Col
        md={6}
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p
          className="font-weight-bold"
        >
          Trace the pseudocode and enter the output that will be produced when the program is run
        </p>
        <pre
          className="text-monospace"
        >
          SUBROUTINE increase_num()
    num2 ← 2
    num1 ← num2 + 5
    num2 ← 13
ENDSUBROUTINE

num1 ← 4
num2 ← 10

increase_num()

num3 ← 4

OUTPUT num1 + num2 + num3
        </pre>
      </Col>
      <Col
        md={6}
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="text-center"
        >
          <img
            alt="Student illustration"
            className="img-fluid"
            src="/assets/ics_spot.svg"
          />
        </div>
      </Col>
    </Row>
  </section>
  <section
    id="events"
  >
    <Row
      tag="div"
    >
      <Col
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <h1
          className="text-center"
        >
          We run events
        </h1>
        <hr />
      </Col>
    </Row>
    <Row
      tag="div"
    >
      <Col
        md={
          Object {
            "offset": 3,
            "size": 6,
          }
        }
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p
          className="text-center"
        >
          We have a number of events throughout the year. Below you can see a snapshot of some of the events we run. On our events page you will see more, and see how you can sign up for more information or sign up to events themselves.
        </p>
      </Col>
    </Row>
    <Row
      tag="div"
    >
      <Col
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <CardDeck
          tag="div"
        >
          <Card
            tag="div"
          >
            <div
              className="text-center"
            >
              <CardImg
                alt="Teacher event illustration"
                src="assets/event-1.png"
                tag="img"
                top={true}
              />
            </div>
            <CardBody
              tag="div"
            >
              <CardTitle
                className="font-weight-bold"
                tag="div"
              >
                Teacher Workshop
              </CardTitle>
              <CardText
                tag="p"
              >
                Embedding isaaccomputerscience in your teaching
              </CardText>
              <CardText
                tag="p"
              >
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    When:
                  </span>
                   
                   
                   Wed 27 Mar 2019
                </span>
                <span
                  className="d-block"
                >
                  12:00 PM — 4:00 PM
                </span>
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    Location:
                  </span>
                   
                   
                   NUSTEM, Northumbria
                </span>
              </CardText>
              <CardText
                tag="p"
              >
                <Link
                  to="event/event1"
                >
                  View Details
                </Link>
              </CardText>
            </CardBody>
          </Card>
          <Card
            tag="div"
          >
            <div
              className="text-center"
            >
              <CardImg
                alt="CPD event illustration"
                src="assets/event-2.png"
                tag="img"
                top={true}
              />
            </div>
            <CardBody
              tag="div"
            >
              <CardTitle
                className="font-weight-bold"
                tag="div"
              >
                Virtual teacher CPD
              </CardTitle>
              <CardText
                tag="p"
              >
                Introduction to teaching with isaaccomputerscience.org
              </CardText>
              <CardText
                tag="p"
              >
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    When:
                  </span>
                   
                   
                   Wed 3 Apr 2019
                </span>
                <span
                  className="d-block"
                >
                  12:00 PM — 13:00 PM
                </span>
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    Location:
                  </span>
                   
                   
                   Online
                </span>
              </CardText>
              <CardText
                tag="p"
              >
                <Link
                  to="event/event1"
                >
                  View Details
                </Link>
              </CardText>
            </CardBody>
          </Card>
          <Card
            tag="div"
          >
            <div
              className="text-center"
            >
              <CardImg
                alt="Teacher event illustration"
                src="assets/event-3.png"
                tag="img"
                top={true}
              />
            </div>
            <CardBody
              tag="div"
            >
              <CardTitle
                className="font-weight-bold"
                tag="div"
              >
                Student Workshop
              </CardTitle>
              <CardText
                tag="p"
              >
                Problem Solving with Vectors
              </CardText>
              <CardText
                tag="p"
              >
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    When:
                  </span>
                   
                   
                   Wed 27 Mar 2019
                </span>
                <span
                  className="d-block"
                >
                  5:30 PM — 7:30 PM
                </span>
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    Location:
                  </span>
                   
                   
                   Abbey College, Cambridge
                </span>
              </CardText>
              <CardText
                tag="p"
              >
                <Link
                  to="event/event1"
                >
                  View Details
                </Link>
              </CardText>
            </CardBody>
          </Card>
        </CardDeck>
      </Col>
    </Row>
  </section>
</div>
`;

exports[`HomepageComponent renders correctly with a null valued user 1`] = `
<div
  id="homepage"
>
  <section
    id="call-to-action"
  >
    <Row
      tag="div"
    >
      <Col
        md={6}
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Row
          tag="div"
        >
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <h1>
              A-level Computer Science Learning
            </h1>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla feugiat lorem nisl, sed convallis dui lobortis volutpat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla feugiat lorem nisl, sed convallis dui lobortis volutpat.
            </p>
          </Col>
        </Row>
        <Row
          tag="div"
        >
          <Col
            size={6}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              block={true}
              color="primary"
              tag={[Function]}
              to="/register"
            >
              Sign up
            </Button>
          </Col>
          <Col
            size={6}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              block={true}
              color="dark"
              outline={true}
              tag={[Function]}
              to="/login"
            >
              Log in
            </Button>
          </Col>
        </Row>
      </Col>
      <Col
        md={6}
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <img
          alt="Students illustration"
          className="img-fluid"
          src="/assets/ics_hero.svg"
        />
      </Col>
    </Row>
  </section>
  <hr />
  <section
    id="why-sign-up"
  >
    <h1
      className="text-center"
    >
      Why sign up?
    </h1>
    <Row
      tag="div"
    >
      <Col
        md={6}
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Card
          tag="div"
        >
          <CardBody
            tag="div"
          >
            <CardTitle
              tag="h2"
            >
              Benefits for
               
              <span
                className="text-primary"
              >
                teachers
              </span>
            </CardTitle>
            <ul>
              <li>
                Easy to set homework
              </li>
              <li>
                Easy to check homework
              </li>
              <li>
                See progress
              </li>
              <li>
                Managing different classes
              </li>
              <li>
                Increase in pass rate
              </li>
              <li>
                Syllabus information
              </li>
              <li>
                Teachers Events
              </li>
            </ul>
          </CardBody>
        </Card>
      </Col>
      <Col
        md={6}
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Card
          tag="div"
        >
          <CardBody
            tag="div"
          >
            <CardTitle
              tag="h2"
            >
              Benefits for
               
              <span
                className="text-primary"
              >
                students
              </span>
            </CardTitle>
            <ul>
              <li>
                Easy to set homework
              </li>
              <li>
                Easy to check homework
              </li>
              <li>
                See progress
              </li>
              <li>
                Managing different classes
              </li>
              <li>
                Increase in pass rate
              </li>
              <li>
                Syllabus information
              </li>
              <li>
                Student Events
              </li>
            </ul>
          </CardBody>
        </Card>
      </Col>
    </Row>
  </section>
  <section
    className="px-5 py-4"
    id="headline-content"
  >
    <h3>
      Featured-question
    </h3>
    <Row
      className="p-1"
      tag="div"
    >
      <Col
        md={6}
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p
          className="font-weight-bold"
        >
          Trace the pseudocode and enter the output that will be produced when the program is run
        </p>
        <pre
          className="text-monospace"
        >
          SUBROUTINE increase_num()
    num2 ← 2
    num1 ← num2 + 5
    num2 ← 13
ENDSUBROUTINE

num1 ← 4
num2 ← 10

increase_num()

num3 ← 4

OUTPUT num1 + num2 + num3
        </pre>
      </Col>
      <Col
        md={6}
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="text-center"
        >
          <img
            alt="Student illustration"
            className="img-fluid"
            src="/assets/ics_spot.svg"
          />
        </div>
      </Col>
    </Row>
  </section>
  <section
    id="events"
  >
    <Row
      tag="div"
    >
      <Col
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <h1
          className="text-center"
        >
          We run events
        </h1>
        <hr />
      </Col>
    </Row>
    <Row
      tag="div"
    >
      <Col
        md={
          Object {
            "offset": 3,
            "size": 6,
          }
        }
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p
          className="text-center"
        >
          We have a number of events throughout the year. Below you can see a snapshot of some of the events we run. On our events page you will see more, and see how you can sign up for more information or sign up to events themselves.
        </p>
      </Col>
    </Row>
    <Row
      tag="div"
    >
      <Col
        size={12}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <CardDeck
          tag="div"
        >
          <Card
            tag="div"
          >
            <div
              className="text-center"
            >
              <CardImg
                alt="Teacher event illustration"
                src="assets/event-1.png"
                tag="img"
                top={true}
              />
            </div>
            <CardBody
              tag="div"
            >
              <CardTitle
                className="font-weight-bold"
                tag="div"
              >
                Teacher Workshop
              </CardTitle>
              <CardText
                tag="p"
              >
                Embedding isaaccomputerscience in your teaching
              </CardText>
              <CardText
                tag="p"
              >
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    When:
                  </span>
                   
                   
                   Wed 27 Mar 2019
                </span>
                <span
                  className="d-block"
                >
                  12:00 PM — 4:00 PM
                </span>
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    Location:
                  </span>
                   
                   
                   NUSTEM, Northumbria
                </span>
              </CardText>
              <CardText
                tag="p"
              >
                <Link
                  to="event/event1"
                >
                  View Details
                </Link>
              </CardText>
            </CardBody>
          </Card>
          <Card
            tag="div"
          >
            <div
              className="text-center"
            >
              <CardImg
                alt="CPD event illustration"
                src="assets/event-2.png"
                tag="img"
                top={true}
              />
            </div>
            <CardBody
              tag="div"
            >
              <CardTitle
                className="font-weight-bold"
                tag="div"
              >
                Virtual teacher CPD
              </CardTitle>
              <CardText
                tag="p"
              >
                Introduction to teaching with isaaccomputerscience.org
              </CardText>
              <CardText
                tag="p"
              >
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    When:
                  </span>
                   
                   
                   Wed 3 Apr 2019
                </span>
                <span
                  className="d-block"
                >
                  12:00 PM — 13:00 PM
                </span>
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    Location:
                  </span>
                   
                   
                   Online
                </span>
              </CardText>
              <CardText
                tag="p"
              >
                <Link
                  to="event/event1"
                >
                  View Details
                </Link>
              </CardText>
            </CardBody>
          </Card>
          <Card
            tag="div"
          >
            <div
              className="text-center"
            >
              <CardImg
                alt="Teacher event illustration"
                src="assets/event-3.png"
                tag="img"
                top={true}
              />
            </div>
            <CardBody
              tag="div"
            >
              <CardTitle
                className="font-weight-bold"
                tag="div"
              >
                Student Workshop
              </CardTitle>
              <CardText
                tag="p"
              >
                Problem Solving with Vectors
              </CardText>
              <CardText
                tag="p"
              >
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    When:
                  </span>
                   
                   
                   Wed 27 Mar 2019
                </span>
                <span
                  className="d-block"
                >
                  5:30 PM — 7:30 PM
                </span>
                <span
                  className="d-block"
                >
                  <span
                    className="font-weight-bold"
                  >
                    Location:
                  </span>
                   
                   
                   Abbey College, Cambridge
                </span>
              </CardText>
              <CardText
                tag="p"
              >
                <Link
                  to="event/event1"
                >
                  View Details
                </Link>
              </CardText>
            </CardBody>
          </Card>
        </CardDeck>
      </Col>
    </Row>
  </section>
</div>
`;
